semver/compare.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+import { SemVer } from "./semver.ts";
+
+/**
+ * Compare two semantic version objects.
+ *
+ * Returns `0` if `v1 == v2`, or `1` if `v1` is greater, or `-1` if `v2` is
+ * greater.
+ *
+ * Sorts in ascending order if passed to `Array.sort()`,
+ * @param s0
+ * @param s1
+ * @returns
+ */
+export function compare(
+  s0: SemVer,
+  s1: SemVer,
+): 1 | 0 | -1 {
+  if (s0 === s1) return 0;
+  return (
+    compareNumber(s0.major, s1.major) ||
+    compareNumber(s0.minor, s1.minor) ||
+    compareNumber(s0.patch, s1.patch) ||
+    checkIdentifier(s0.prerelease, s1.prerelease) ||
+    compareIdentifier(s0.prerelease, s1.prerelease) ||
+    checkIdentifier(s1.build, s0.build) ||
+    compareIdentifier(s0.build, s1.build)
```

Comment:
Comparing `build` by default feels wrong to me as the spec says the below

> Build metadata MUST be ignored when determining version precedence. Thus two versions that differ only in the build metadata, have the same precedence.

https://semver.org/#spec-item-10

I think we should keep `compareBuild` API to support the case where the users want to compare `build` as well